WcaOnRails/app/models/competition.rb

```diff
@@ -96,24 +101,10 @@ def info
     info
   end
 
-  before_validation :clone_competition, on: [:create]
-  def clone_competition
-    if competition_id_to_clone.present?
-      competition_to_clone = Competition.find_by_id(competition_id_to_clone)
-      if competition_to_clone
-        attributes = competition_to_clone.attributes
-        # Don't clone id, name, cellName, and dates.
-        %w(id name cellName year month day endMonth endDay registration_open registration_close results_posted_at).each { |attribute| attributes.delete attribute }
-        # Make sure the new competition is not publicly visible and is open for
-        # editing.
-        attributes["showAtAll"] = false
-        attributes["isConfirmed"] = false
-        assign_attributes(attributes)
-        self.organizers |= competition_to_clone.organizers
-        self.delegates |= competition_to_clone.delegates
-      else
-        errors.add(:competition_id_to_clone, "invalid")
-      end
+  def build_clone
+    Competition.new(attributes.except(*UNCLONEABLE_ATTRIBUTES)).tap do |clone|
```

Comment:
As you've added both attributes groups as Lars suggested, it would be better to do
```ruby
attributes.slice(CLONABLE_ATTRIBUTES)
```

This really cleaned things up.